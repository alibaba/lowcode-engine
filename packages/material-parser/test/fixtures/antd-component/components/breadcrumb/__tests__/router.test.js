import React from 'react';
import { Route, Switch, Link, withRouter, MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import Breadcrumb from '../index';

const Apps = () => (
  <ul className="app-list">
    <li>
      <Link to="/apps/1">Application1</Link>：<Link to="/apps/1/detail">Detail</Link>
    </li>
    <li>
      <Link to="/apps/2">Application2</Link>：<Link to="/apps/2/detail">Detail</Link>
    </li>
  </ul>
);

const breadcrumbNameMap = {
  '/apps': 'Application List',
  '/apps/1': 'Application1',
  '/apps/2': 'Application2',
  '/apps/1/detail': 'Detail',
  '/apps/2/detail': 'Detail',
};

describe('react router', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  // https://github.com/airbnb/enzyme/issues/875
  it('react router 4', () => {
    if (process.env.REACT === '15') {
      return;
    }
    const Home = withRouter(props => {
      const { location, history } = props;
      const pathSnippets = location.pathname.split('/').filter(i => i);
      const extraBreadcrumbItems = pathSnippets.map((_, index) => {
        const url = `/${pathSnippets.slice(0, index + 1).join('/')}`;
        return (
          <Breadcrumb.Item key={url}>
            <Link to={url}>{breadcrumbNameMap[url]}</Link>
          </Breadcrumb.Item>
        );
      });
      const breadcrumbItems = [
        <Breadcrumb.Item key="home">
          <Link to="/">Home</Link>
        </Breadcrumb.Item>,
      ].concat(extraBreadcrumbItems);
      return (
        <div className="demo">
          <div className="demo-nav">
            <a onClick={() => history.push('/')}>Home</a>
            <a onClick={() => history.push('/apps')}>Application List</a>
          </div>
          <Switch>
            <Route path="/apps" component={Apps} />
            <Route render={() => <span>Home Page</span>} />
          </Switch>
          <Breadcrumb>{breadcrumbItems}</Breadcrumb>
        </div>
      );
    });
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <Home />
      </MemoryRouter>,
    );
    expect(wrapper.find('BreadcrumbItem').length).toBe(1);
    expect(
      wrapper
        .find('BreadcrumbItem .ant-breadcrumb-link')
        .at(0)
        .text(),
    ).toBe('Home');
    wrapper
      .find('.demo-nav a')
      .at(1)
      .simulate('click');
    expect(wrapper.find('BreadcrumbItem').length).toBe(2);
    expect(
      wrapper
        .find('BreadcrumbItem .ant-breadcrumb-link')
        .at(1)
        .text(),
    ).toBe('Application List');
  });

  it('react router 3', () => {
    const routes = [
      {
        name: 'home',
        breadcrumbName: 'Home',
        path: '/',
        childRoutes: [
          {
            name: 'apps',
            breadcrumbName: 'Application List',
            path: 'apps',
            childRoutes: [
              {
                name: 'app',
                breadcrumbName: 'Application:id',
                path: ':id',
                childRoutes: [
                  {
                    name: 'detail',
                    breadcrumbName: 'Detail',
                    path: 'detail',
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        name: 'apps',
        breadcrumbName: 'Application List',
        path: 'apps',
        childRoutes: [
          {
            name: 'app',
            breadcrumbName: 'Application:id',
            path: ':id',
            childRoutes: [
              {
                name: 'detail',
                breadcrumbName: 'Detail',
                path: 'detail',
              },
            ],
          },
        ],
      },
      {
        name: 'app',
        breadcrumbName: 'Application:id',
        path: ':id',
        childRoutes: [
          {
            name: 'detail',
            breadcrumbName: 'Detail',
            path: 'detail',
          },
        ],
      },
      {
        name: 'detail',
        breadcrumbName: 'Detail',
        path: 'detail',
      },
    ];
    const wrapper = mount(<Breadcrumb routes={routes} params={{ id: 1 }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
