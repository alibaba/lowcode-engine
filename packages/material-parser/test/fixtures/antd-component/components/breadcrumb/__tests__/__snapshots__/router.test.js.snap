// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react router react router 3 1`] = `
<Breadcrumb
  params={
    Object {
      "id": 1,
    }
  }
  routes={
    Array [
      Object {
        "breadcrumbName": "Home",
        "childRoutes": Array [
          Object {
            "breadcrumbName": "Application List",
            "childRoutes": Array [
              Object {
                "breadcrumbName": "Application:id",
                "childRoutes": Array [
                  Object {
                    "breadcrumbName": "Detail",
                    "name": "detail",
                    "path": "detail",
                  },
                ],
                "name": "app",
                "path": ":id",
              },
            ],
            "name": "apps",
            "path": "apps",
          },
        ],
        "name": "home",
        "path": "/",
      },
      Object {
        "breadcrumbName": "Application List",
        "childRoutes": Array [
          Object {
            "breadcrumbName": "Application:id",
            "childRoutes": Array [
              Object {
                "breadcrumbName": "Detail",
                "name": "detail",
                "path": "detail",
              },
            ],
            "name": "app",
            "path": ":id",
          },
        ],
        "name": "apps",
        "path": "apps",
      },
      Object {
        "breadcrumbName": "Application:id",
        "childRoutes": Array [
          Object {
            "breadcrumbName": "Detail",
            "name": "detail",
            "path": "detail",
          },
        ],
        "name": "app",
        "path": ":id",
      },
      Object {
        "breadcrumbName": "Detail",
        "name": "detail",
        "path": "detail",
      },
    ]
  }
  separator="/"
>
  <div
    className="ant-breadcrumb"
  >
    <BreadcrumbItem
      key="Home"
      separator="/"
    >
      <span>
        <span
          className="ant-breadcrumb-link"
        >
          <a
            href="#/"
          >
            Home
          </a>
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
    </BreadcrumbItem>
    <BreadcrumbItem
      key="apps"
      separator="/"
    >
      <span>
        <span
          className="ant-breadcrumb-link"
        >
          <a
            href="#/apps"
          >
            Application List
          </a>
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
    </BreadcrumbItem>
    <BreadcrumbItem
      key="1"
      separator="/"
    >
      <span>
        <span
          className="ant-breadcrumb-link"
        >
          <a
            href="#/apps/1"
          >
            Application1
          </a>
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
    </BreadcrumbItem>
    <BreadcrumbItem
      key="detail"
      separator="/"
    >
      <span>
        <span
          className="ant-breadcrumb-link"
        >
          <span>
            Detail
          </span>
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
    </BreadcrumbItem>
  </div>
</Breadcrumb>
`;
